src/openMVG/localization/VoctreeLocalizer.cpp

```diff
@@ -1038,13 +1040,410 @@ bool VoctreeLocalizer::robustMatching(matching::RegionsMatcherT<MatcherT> & matc
+  }
+
+  std::vector<std::map< pair<IndexT, IndexT>, std::size_t > > vec_occurrences(numCams);
+  std::vector<std::vector<voctree::DocMatch> > vec_matchedImages(numCams);
+  std::vector<Mat> vec_pts3D(numCams);
+  std::vector<Mat> vec_pts2D(numCams);
+
+  // for each camera retrieve the associations
+  //@todo parallelize?
+  size_t numAssociations = 0;
+  for(std::size_t cam = 0; cam < numCams; ++cam)
+  {
+
+    // this map is used to collect the 2d-3d associations as we go through the images
+    // the key is a pair <Id3D, Id2d>
+    // the element is the pair 3D point - 2D point
+    auto &occurrences = vec_occurrences[cam];
+    auto &matchedImages = vec_matchedImages[cam];
+    auto &imageSize = vec_imageSize[cam];
+    Mat &pts3D = vec_pts3D[cam];
+    Mat &pts2D = vec_pts2D[cam];
+    cameras::Pinhole_Intrinsic_Radial_K3 &queryIntrinsics = vec_queryIntrinsics[cam];
+    features::SIFT_Regions &queryRegions = *dynamic_cast<features::SIFT_Regions*> (vec_queryRegions[cam].get());
+    const bool useInputIntrinsics = true;
+    getAllAssociations(queryRegions,
+                       imageSize,
+                       *param,
+                       useInputIntrinsics,
+                       queryIntrinsics,
+                       occurrences,
+                       pts2D,
+                       pts3D,
+                       matchedImages);
+    numAssociations += occurrences.size();
+  }
+  
+  // @todo Here it could be possible to filter the associations according to their
+  // occurrences, eg giving priority to those associations that are more frequent
+
+  const std::size_t minNumAssociations = 5;  //possible parameter?
+  if(numAssociations < minNumAssociations)
+  {
+    POPART_COUT("[poseEstimation]\tonly " << numAssociations << " have been found, not enough to do the resection!");
+    return false;
+  }
+  
+  {
+    for(std::size_t cam = 0; cam < vec_subPoses.size(); ++cam)
+    {
+    POPART_COUT("Rotation: " << vec_subPoses[cam].rotation());
+    POPART_COUT("Centre: " << vec_subPoses[cam].center());
+    }
+  }
+  
+  std::vector<std::vector<std::size_t> > vec_inliers;
+  const bool resectionOk = rigResection(vec_pts2D,
+                                        vec_pts3D,
+                                        vec_queryIntrinsics,
+                                        vec_subPoses,
+                                        rigPose,
+                                        vec_inliers);
+  
+  if(!resectionOk)
+  {
+    for(std::size_t cam = 0; cam < numCams; ++cam)
+    {
+      // empty result with isValid set to false
+      vec_locResults.emplace_back();
+    }
+    return resectionOk;
+  }
+  
+  { // just debugging stuff, this block can be removed
+    
+    if(vec_inliers.size() < numCams)
+    {
+      // in general the inlier should be spread among different cameras
+      POPART_CERR("****************** Oh, this is weird!!!!");
+    }
+    else
+    {
+      for(std::size_t cam = 0; cam < numCams; ++cam)
```

Comment:
`for(std::size_t cam = 0; cam < vec_inliers.size(); ++cam)`
So you can always print it.